01/Table.js

```diff
@@ -2,43 +2,66 @@ import React from 'react';
 import TableHeader from './components/TableHeader.js'
 import TableBody from './components/TableBody.js'
 import TableFooter from './components/TableFooter.js';
+import { v4 as uuid } from 'uuid';
 
 export default class Table extends React.Component {
     render() {
         return (
             <table className="table">
                 <caption className="title">Products Info</caption>
-                <TableHeader columnNames={this.setColumnNames()} />
-                <TableBody bodyContent={this.setBodyContent()} />
-                <TableFooter totalPrice={this.setTotalPrice()} />
+                <TableHeader columnData={this.setColumnData()} />
+                <TableBody bodyData={this.setBodyData()} />
+                <TableFooter footerData={this.setFooterData()} />
             </table>
         )
     }
 
-    setColumnNames = () => {
+    setColumnData = () => {
         const { data } = this.props;
         const { id , ...names } = data[0];
 
-        return [...Object.keys(names), 'total price'];
+        const columnNames = [...Object.keys(names) , 'total price'];
+
+        const columnData = columnNames.map( name => {
+            const nameId = uuid();
+            return [ name , nameId ];
+        });
```

Comment:
Zdecydowanie lepiej! Tylko jeśli masz "name" i jesteś pewny, że to będzie unikalna nazwa (a tutaj pewnie tak bedzie) to ja bym z niej skorzystał zamiast kombinować z `uuid()`